### PR TITLE
GH-3203 support optionalNamespaces

### PR DESCRIPTION
**Pull request by DmitryBespalov**
Handles #3203 

Changes proposed in this pull request:
- Updated WalletConnectSwiftV2 library to the newest version
- Evolved `DummySignerFactory` into NullCryptoProvider to fix breaking changes
- Refactored `approveSession` method to support only safe's chain. Please see link for the reason behind this recommended approach: https://safe-global.notion.site/safe-global/WalletConnect-v2-update-Issues-and-solutions-for-smart-wallets-3fc32fad6af4485fa5823eaebd486819

### CODE REVIEW

File: Multisig.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -7172,8 +7172,8 @@
 			isa = XCRemoteSwiftPackageReference;
 			repositoryURL = "https://github.com/WalletConnect/WalletConnectSwiftV2";
 			requirement = {
-				branch = main;
-				kind = branch;
+				kind = upToNextMajorVersion;
```
Comment (biafra23) [inline Multisig.xcodeproj/project.pbxproj:7175]: Thank you!

File: Multisig/UI/WalletConnectV2/WalletConnectManager.swift [hunk 2]
```diff
@@ -153,30 +153,51 @@ class WalletConnectManager {
...
+    //   - if no such chain found in proposal, connection will fail
     func approveSession(proposal: Session.Proposal) {
         Task {
             guard let safe = try? Safe.getSelected() else { return }
-            guard isValidProposal(proposal: proposal, safe: safe) else {
+                        
+            // Step 1: find a compatible namespace with safe's chain
+            var resultNamespaces = [String: SessionNamespace]()
+
+            let EVM_COMPATIBLE_NETWORK = "eip155"
```
Comment (biafra23) [inline Multisig/UI/WalletConnectV2/WalletConnectManager.swift:169]: Why not reuse the static constant EVM_COMPATIBLE_NETWORK created above?

**Review by biafra23 — APPROVED** [review]
LGTM 🚀

**Review by MouazAlzahabi — APPROVED** [review]

File: Multisig/UI/WalletConnectV2/WalletConnectManager.swift (hunk 2 above)
Comment (DmitryBespalov) [reply Multisig/UI/WalletConnectV2/WalletConnectManager.swift:169]: I wanted to put the constant / var declaration closer to its usage. And I didn't remove the static variable

**Pull request merged**